cpp/src/gandiva/tests/generate_data.h

```diff
@@ -46,7 +46,11 @@ class BoundedInt32DataGenerator : public Int32DataGenerator {
   explicit BoundedInt32DataGenerator(uint32_t upperBound)
       : Int32DataGenerator(), upperBound_(upperBound) {}
 
-  int32_t GenerateData() { return (rand_r(&seed_) % upperBound_); }
+  int32_t GenerateData() {
+    int32_t value = (rand_r(&seed_) % upperBound_);
```

Comment:
Why don't you use [C++ random number generation](https://en.cppreference.com/w/cpp/numeric/random) in this header?

Reply:
sure changed.

curious is c++ generator better in some way? (could not find in a quick google search).

Reply:
It probably depends on the libc. It provides many more options, though.